### PR TITLE
13 google auth and manual signin

### PR DESCRIPTION
**Pull request by Saeed99Madi**

### CODE REVIEW

**Review by mohmmed23 — DISMISSED** [review]

**Review by AseelL — APPROVED** [review]

File: src/config/auth.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+const GoogleStrategy = GooglePassport.Strategy
+
+passport.serializeUser(function (user: any, done) {
+  done(null, user.id)
+})
+
+passport.deserializeUser((obj: any, done) => {
+  done(null, obj)
+})
```
Comment (aaamra) [inline src/config/auth.ts:19]: delete

File: src/config/auth.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+    const verified = await bcrypt.compare(password, user.password)
+    verified
+      ? done(null, user)
+      : done(null, false, { message: 'password not match' })
+  } catch (error) {
+    done(error)
+  }
+}))
+
```
Comment (aaamra) [inline src/config/auth.ts:46]: we dont need local strategy

File: src/config/auth.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  passReqToCallback: true,
+  scope: ['profile', 'email']
+},
+async (req:Request, accessToken:any, refreshToken:any, profile:any, done:any) => {
+  try {
+    const user = await findOrCreate(profile.displayName, profile.emails[0].value, profile.photos[0].value)
+
+    if (!user) {
+      console.log('can not found user')
+      return done(null, false, { message: 'can not found user' })
```
Comment (aaamra) [inline src/config/auth.ts:60]:
```suggestion
      return done(null, false)
```

File: src/config/auth.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+  new JWTStrategy.Strategy(
+    {
+      jwtFromRequest: (req:Request) => {
+        let token = null
+        if (req && req.cookies) {
+          token = req.cookies.jwt
+          console.log(token, 'Token')
+        }
+        return token
+      },
```
Comment (aaamra) [inline src/config/auth.ts:81]: we need to extract the token from headers

File: src/config/environment.ts [hunk 5]
```diff
@@ -3,8 +3,16 @@ import dotenv from 'dotenv'
 dotenv.config()
 
 const port = process.env.PORT ?? 8080
-let dbUrl: string | undefined
+
+const googleClientId = process.env.GOOGLE_CLIENT_ID ?? ''
+
```
Comment (aaamra) [inline src/config/environment.ts:8]:
```suggestion
const googleClientId = process.env.GOOGLE_CLIENT_ID
if(!googleClientId) {
 throw new Error('invalid google client id');
}
```

**Review by aaamra — CHANGES_REQUESTED** [review]

**Review by aaamra — DISMISSED** [review]

**Review by AseelL — APPROVED** [review]

**Review by salsabeelomar — APPROVED** [review]

File: __tests__/signIn.test.ts [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+  test('incorrect password', (done) => {
+    supertest(router)
+      .post('/signin')
+      .send({
+        email: 'saeed@gmail.com',
+        password: '123asa456'
+      })
+      .expect(422)
+      .expect('Content-Type', /html/)
+      .end((err:any, res: any) => {
```
Comment (mohmmed23) [inline __tests__/signIn.test.ts:19]: you should change ```any``` to interface related data type to res and err

**Review by mohmmed23 — CHANGES_REQUESTED** [review]

**Review by salsabeelomar — DISMISSED** [review]

**Review by aaamra — APPROVED** [review]

**Review by AseelL — APPROVED** [review]

**Review by mohmmed23 — APPROVED** [review]

**Review by salsabeelomar — APPROVED** [review]

**Pull request merged**